lcm/lcm-cpp.hpp

```diff
@@ -524,12 +524,16 @@ class Subscription {
     friend class LCM;
 
   protected:
-    Subscription(){};
+    Subscription() { channel.reserve(LCM_MAX_CHANNEL_NAME_LENGTH); };
+
     /**
      * The underlying lcm_subscription_t object wrapped by this
      * subscription.
      */
     lcm_subscription_t *c_subs;
+
+    // The channel that `this` is subscribed to.
+    std::string channel;
```

Comment:
Suggest renaming to `channel_buf` or something like that to indicate that it's really a working buffer that get's overwritten every time a new message is handled, and also adding comments to that effect.